Enterprise-CMCS/macpro-onemac
# clear table state if not authenticated (upon redirect to login page)

Story: https://qmacbis.atlassian.net/browse/OY2-23899
Endpoint: See github-actions bot comment

### Details

Currently table filters (state) is cleared upon a timeout and an intentional logout action; however it was not being cleared upon a token timeout if the user did not have the page loaded at the time. This solution will cause the state to be cleared on any route change (such as on app load if user uses a direct url navigation or reopens the app from start) by weaving into the logic already built into our routing for unauthenticated users.

### Changes

- updated routing logic for when an unauthenticated user (including expired token) loads a route then the table state will be cleared

### Test Plan

1. Login as any user
2. Set table filters
3. close tab and wait for token to expire
4. Open new tab and you should be redirected to login page
5. login again and verify table state is reset
